share/availability/discovery/discovery.go

```diff
@@ -283,14 +286,17 @@ func (d *Discovery) discover(ctx context.Context) bool {
 		drainChannel(ticker.C)
 		select {
 		case <-findCtx.Done():
+			d.metrics.observeFindPeers(true, d.set.Size() >= d.set.Limit())
 			return true
```

Comment:
A bit weird that we return true, but for the metric, we still do the comparison. Either we should emit true for both, or we do the comparison for both. I vote the former as we know at this point that it's true 100%

Reply:
Yeah, just rebasing artifact. Should be (true,true)